### DISCUSSION TITLE
Troubleshooting cbs_parse_data

### DISCUSSION THREAD

**Discussion by mike-battista** [TheOwenLab/cbs-toolkit#5] [Q&A] [answered]
Hey @conorwild, hope I'm putting this in the right place.

I think I've gotten as far as installing Docker and accessing the CBS Python image. Now I'm trying to use cbs_parse_data. Running the help command works fine, but with a real data file, I'm trying:

`docker run cbs_parse_data CBS_Research_Example_Data_V2.csv parse_example
`

And I get:

```
Unable to find image 'cbs_parse_data:latest' locally
docker: Error response from daemon: pull access denied for cbs_parse_data, repository does not exist or may require 'docker login': denied: requested access to the resource is denied.
```

Any idea what could be going on? I'm in the directory with the file, which I've tried with and without spaces. With the access token thing, everything seemed to go okay and it said I was logged in.

Thanks so much. Excited to play with this once I get it working!

**Comment by conorwild** [comment, marked as answer]
Hey @mike-battista, it looks like you missing some of the Docker "magic" from your command. Try this:
```
docker run --rm -it -v $PWD:/tmp -w /tmp ghcr.io/theowenlab/cbspython:latest cbs_parse_data CBS_Research_Example_Data_V2.csv parse_example 
```

The error message you posted says that it is trying to find the `cbs_parse_data:latest` image - but the image is called `cbspython:latest`, and `cbs_parse_data` is the name of the script to execute within that container. Give that a shot!

**Reply by mike-battista** [reply]
I finally got around to trying this, and it worked for me too! Thanks so much @conorwild .

**Comment by SajeevanS** [comment]
Hey Conor, 
So I'm trying out the cbs_parse_data line and running into errors. I run the above code changing my input and output file names, and I run into the following error below. 

Any thoughts? 

`WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
Traceback (most recent call last):
  File "/opt/venv/bin/cbs_parse_data", line 33, in <module>
    sys.exit(load_entry_point('cbspython==0.1.0', 'console_scripts', 'cbs_parse_data')())
  File "/opt/venv/lib/python3.9/site-packages/cbspython/scripts/parse_raw_data.py", line 95, in main
    report = cbs.Report(
  File "/opt/venv/lib/python3.9/site-packages/cbspython/cbsreport.py", line 263, in __init__
    raise TypeError(f"Not sure how to handle {src}")
TypeError: Not sure how to handle CBS_Data`

**Reply by conorwild** [reply]
Hey @SajeevanS , you can ignore the warning for now (`WARNING: the requested image's platform...`). 
Good news: it runs!
Good news: it's trying to open a file.
Bad news: it's not sure what kind of file you are trying to load. You need to use the full filename of your CBS data export, and it should end with `.csv` or `.csv.bz2`. Anyways, that's what I think the issue is, but I can't tell without seeing the command exactly as you tried to run it. Can you copy / paste more of the terminal output, incuding what your command looked like?

**Reply by SajeevanS** [reply]
MacBook-Pro-102:Desktop Sajeevan$ docker run --rm -it -v $PWD:/tmp -w /tmp ghcr.io/theowenlab/cbspython:latest cbs_parse_data CBS_Data CBS_Data_clean
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
Traceback (most recent call last):
  File "/opt/venv/bin/cbs_parse_data", line 33, in <module>
    sys.exit(load_entry_point('cbspython==0.1.0', 'console_scripts', 'cbs_parse_data')())
  File "/opt/venv/lib/python3.9/site-packages/cbspython/scripts/parse_raw_data.py", line 95, in main
    report = cbs.Report(
  File "/opt/venv/lib/python3.9/site-packages/cbspython/cbsreport.py", line 263, in __init__
    raise TypeError(f"Not sure how to handle {src}")
TypeError: Not sure how to handle CBS_Data

**Reply by SajeevanS** [reply]
my file name is CBS_Data and I want it to name it as CBS_Data_clean

**Reply by conorwild** [reply]
Is it a CSV? Then you should use `CBS_Data.csv`. If it doesn't have that extension, then you should rename it to have that.

**Reply by SajeevanS** [reply]
I see! I think i'm making progress, but having some issues with setting up my directory. 

MacBook-Pro-102:~ Sajeevan$ docker run --rm -it -v $PWD:/tmp -w /tmp ghcr.io/theowenlab/cbspython:latest cbs_parse_data CBS_Data.csv  CBS_Data_clean
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
Traceback (most recent call last):
  File "/opt/venv/bin/cbs_parse_data", line 33, in <module>
    sys.exit(load_entry_point('cbspython==0.1.0', 'console_scripts', 'cbs_parse_data')())
  File "/opt/venv/lib/python3.9/site-packages/cbspython/scripts/parse_raw_data.py", line 95, in main
    report = cbs.Report(
  File "/opt/venv/lib/python3.9/site-packages/cbspython/cbsreport.py", line 259, in __init__
    self.data = pd.read_csv(src, delimiter=delimiter)
  File "/opt/venv/lib/python3.9/site-packages/pandas/util/_decorators.py", line 311, in wrapper
    return func(*args, **kwargs)
  File "/opt/venv/lib/python3.9/site-packages/pandas/io/parsers/readers.py", line 680, in read_csv
    return _read(filepath_or_buffer, kwds)
  File "/opt/venv/lib/python3.9/site-packages/pandas/io/parsers/readers.py", line 575, in _read
    parser = TextFileReader(filepath_or_buffer, **kwds)
  File "/opt/venv/lib/python3.9/site-packages/pandas/io/parsers/readers.py", line 934, in __init__
    self._engine = self._make_engine(f, self.engine)
  File "/opt/venv/lib/python3.9/site-packages/pandas/io/parsers/readers.py", line 1218, in _make_engine
    self.handles = get_handle(  # type: ignore[call-overload]
  File "/opt/venv/lib/python3.9/site-packages/pandas/io/common.py", line 786, in get_handle
    handle = open(
FileNotFoundError: [Errno 2] No such file or directory: 'CBS_Data.csv'

**Reply by conorwild** [reply]
OK, first I notice that you are in your "home" directory (`MacBook-Pro-102:~`) and that might not be where your data is located. You need to change directory in your terminal to the directory where your CBS data export file is located. For example, if you put it on your desktop then you would `cd ~/Desktop`. Or, if it's in a subdirectory of your Documents folder called "CBS" then you would `cd ~/Documents/CBS/`. You have to run this in the directory where your data is located.

**Reply by SajeevanS** [reply]
IT WORKED!!! Thank you so much Conor!!! It definitely is a trial and error and a learning experience!! I got my output!
